route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -142,23 +206,167 @@ namespace route_following_plugin
             else
             {
                 ROS_DEBUG_STREAM("Lanechange NOT Needed ");
-                maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, start_speed, target_speed_in_lanelet, route_shortest_path[shortest_path_index].id()));
+                maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, start_speed, target_speed_in_lanelet, { route_shortest_path[shortest_path_index].id() } ));
             }
         }
-        //TO DO  - Update this block so that Stop and wait is the last maneuver ----
-        //add lane follow as last maneuver if there is a lanelet unplanned for in path
+
+        // Add stop and wait maneuver as last maneuver if there is a lanelet unplanned for in path
         if (shortest_path_index < route_shortest_path.size())
         {
-            double target_speed_in_lanelet = findSpeedLimit(route_shortest_path.back());
-            start_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().front()).downtrack;
-            end_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().back()).downtrack;
-            maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, start_speed, target_speed_in_lanelet, route_shortest_path.back().id()));
+
+            // Compute target deceleration for stopping
+            double stopping_accel_limit = accel_limit_ * stopping_accel_limit_multiplier_;
+
+            // Estimate the entry speed for the stopping maneuver
+            double stopping_entry_speed = maneuvers.empty() ? findSpeedLimit(route_shortest_path.back()) : getManeuverEndSpeed(maneuvers.back());
```

Comment:
If there are no maneuvers planned, should we be using the speed limit? Will that not cause stop and wait to increase the speed of the vehicle and then slow it down?

Reply:
Hmm that's a good point. 
I think the current speed would make more sense here. Do you agree?

Reply:
Yes, current speed makes sense here. I guess the assumption there would be that this maneuver will be executed now instead of in the future, but that is in keeping with the fact that the maneuver list is empty before his